### PR TITLE
Syntax checking fixes

### PR DESCRIPTION
**Pull request by RobertBColton**
This pull request addresses several issues with syntax checking. First, it fixes formatting issues in the files modified. Second, it stops an infinite loop when the syntax checking method is called on empty code.

Third, it fixes a regressive logic error introduced in https://github.com/enigma-dev/enigma-dev/commit/e624172e6cfd2bb56824e9f49577ed16145f1212 which caused the minimum argument count of all overloads to be calculated incorrectly. This was fixed by simply initializing min to SIZE_MAX.

Finally, I deleted old garbage from r4 and updated the CBP to no longer include them.

This ticket fully addresses #931

### CODE REVIEW

File: CompilerSource/syntax/syntax.cpp [hunk 1]
```diff
@@ -220,26 +226,23 @@ namespace syncheck
...
-
     pt pos = 0;
     unsigned mymacroind = 0;
     macro_stack_t mymacrostack;
-    lex.clear(); 
+    lex.clear();
     lex.push_back(token(TT_IMPLICIT_SEMICOLON, ";", superPos, 0, true, false, false, mymacroind));
-          
-    
-    syerr = "No error";
```
Comment (JoshDreamland) [inline CompilerSource/syntax/syntax.cpp]: What was this disrupting?

Comment (RobertBColton) [inline CompilerSource/syntax/syntax.cpp]: Nothing, look just a few lines up, I moved it to just before the empty code check.

**Comment by JoshDreamland** [comment]
In the future, please don't mix functionality changes with whitespace corrections. They ruin the diff.

**Comment by RobertBColton** [comment]
I realize that now, I will make an honest effort in the future to do whitespace corrections in a single commit.

**Pull request merged**